eclipse/che-che4z-lsp-for-cobol
# Upgrade Cypress to 6.4.0. Add mochawesome reporter. Add configs.

This PR fixes #765 and #761.  
## How to test
Probably remove `node_modules` and `yarn.lock`. Perform `yarn install`. 
Then run `npm run ts:build`. 
Simply run `npm run cy:run:all:headless` or `npm run cy:run:all:headless -- --config baseUrl=https://theia`. 
After finishing the test run perform `npm run generate:html:report`. This will generate the final HTML report. See the example in the attachments. 

## Results
```
====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔️  LSP\copybook.spec.js                     03:28       20       16        -        4        -│
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔️  LSP\F101836.spec.js                      00:10        1        1        -        -         │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔️  LSP\F102470.spec.js                      00:15        3        2        -        1        -│
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔️  LSP\F95833.spec.js                       00:33        5        4        -        1        -│
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔️  LSP\F95834.spec.js                       00:09        1        1        -        -        -│
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖️  LSP\F96117.spec.js                       02:21        9        5        1        3        -│
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔️  LSP\F96125.spec.js                       02:40        5        5        -        -        -│
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖️  LSP\F97476.spec.js                       03:50       10        3        1        6        -│
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔️  LSP\lsp.spec.js                          05:15       23       20        -        3        -│
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖️  2 of 9 failed (22%)                      18:44       77       57        2       18        -
    
```
HTML report:
[final_report.zip](https://github.com/eclipse/che-che4z-lsp-for-cobol/files/5987373/final_report.zip)

Signed-off-by: Maryna Nalbandian <maryna.nalbandian@broadcom.com>
